tools/external_ic.F90

```diff
@@ -2474,8 +2531,8 @@ subroutine get_fv_ic( Atm, fv_domain, nq )
 ! Read in lat-lon FV core restart file
       fname = Atm%flagstruct%res_latlon_dynamics
 
-      if( file_exist(fname) ) then
-          call field_size(fname, 'T', tsize, field_found=found)
+      if( file_exists(fname) .and. open_file(Latlon_dyn, fname, "read", is_restart=.true.) ) then
```

Comment:
I think you can get rid of the `file_exists`, if the file does not exists then open_file is never going to be true.